.github/workflows/artifacts.yml

```diff
@@ -15,19 +15,17 @@ jobs:
     name: Artifacts generation
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```

Comment:
nit: Because there's a lot of these. Do you think we can make the comment have a link to the commit itself? This way is easier to verify the hash points to the right version for the reviewewers

Reply:
My concern with that is that update-bots (like Dependabot, Renovate) won't increment the commit in the comment, they'll only increment the commit in the `@[...]` and the version in the `# v[...]`.

Reply:
okay, then fine with me

Reply:
Do you think is possible that dependabot does the version changes, and this PR does the other modifications (to not mix the two things)

Think about this. imagine you are an evil person 😈
You might add the hash of an older version with a known exploit, and because these changes are hard to review manually, we will likely miss that in our review